python/cudf/dataframe/index.py

```diff
@@ -601,8 +601,14 @@ def as_index(arbitrary, name=None):
     else:
         if hasattr(arbitrary, 'name') and name is None:
             name = arbitrary.name
-        if len(arbitrary) == 0:
-            return RangeIndex(0, 0, name=name)
+        try:
+            if len(arbitrary) == 0:
+                return RangeIndex(0, 0, name=name)
+        except(TypeError):
+            # except...pass used here v/s checking for __len__ attr
+            # because numpy scalars have a __len__ attribute but
+            # raise TypeError when len() is called on them
+            pass
```

Comment:
I'm not a fan of the `except: pass` pattern, it is extremely dangerous. This would hide any `TypeError` thrown by the `RangeIndex` constructor and cause really difficult to track errors.

Reply:
What about doing `hasattr(arbitrary, len)` - does that have the same issue you mentioned?

Reply:
Yes. Using `if hasattr(arbitrary, len)` will allow numpy scalars, but then actually calling `len()` on them will fail:

```python
In [2]: a = np.array(1)                                                                                                                                                                                         

In [3]: hasattr(a, '__len__')                                                                                                                                                                                   
Out[3]: True

In [4]: len(a)                                                                                                                                                                                                  
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-1a2e6ec5f1e3> in <module>
----> 1 len(a)

TypeError: len() of unsized object
```

I agree that `except...pass` is dangerous and I'll try to think of a better solution.

Reply:
We do a lot of `isinstance()` on this project, that'd be my next approach.